lib/Conversion/GPUXToLLVM/GPUXToLLVMPass.cpp

```diff
@@ -0,0 +1,721 @@
+                  gpux::DestroyStreamOp::Adaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+    auto res =
+        streamDestroyCallBuilder.create(loc, rewriter, adaptor.getGpuxStream());
+    rewriter.replaceOp(op, res.getResults());
+    return mlir::success();
+  }
+};
+
+class GPUXToLLVMPass : public ::imex::ConvertGPUXToLLVMBase<GPUXToLLVMPass> {
+public:
+  explicit GPUXToLLVMPass() {}
+  void runOnOperation() override;
+};
+
+void GPUXToLLVMPass::runOnOperation() {
+  mlir::MLIRContext &context = getContext();
+  mlir::LLVMTypeConverter converter(&context);
+  mlir::RewritePatternSet patterns(&context);
+  mlir::LLVMConversionTarget target(context);
+
+  mlir::arith::populateArithToLLVMConversionPatterns(converter, patterns);
+  mlir::cf::populateControlFlowToLLVMConversionPatterns(converter, patterns);
+  mlir::populateVectorToLLVMConversionPatterns(converter, patterns);
+  mlir::populateMemRefToLLVMConversionPatterns(converter, patterns);
+  mlir::populateFuncToLLVMConversionPatterns(converter, patterns);
+  mlir::populateAsyncStructuralTypeConversionsAndLegality(converter, patterns,
+                                                          target);
+
+  mlir::populateGpuToLLVMConversionPatterns(
+      converter, patterns, mlir::gpu::getDefaultGpuBinaryAnnotation());
+
+  populateControlFlowTypeConversionRewritesAndTarget(converter, patterns,
+                                                     target);
+
+  populateGpuxToLLVMPatternsAndLegality(converter, patterns, target);
+
+  auto mod = getOperation();
+  if (mlir::failed(
+          mlir::applyPartialConversion(mod, target, std::move(patterns))))
+    signalPassFailure();
+}
+
+void populateGpuxToLLVMPatternsAndLegality(mlir::LLVMTypeConverter &converter,
+                                           mlir::RewritePatternSet &patterns,
+                                           mlir::ConversionTarget &target) {
+  auto context = patterns.getContext();
+  auto llvmPointerType =
+      mlir::LLVM::LLVMPointerType::get(mlir::IntegerType::get(context, 8));
+  converter.addConversion([llvmPointerType](gpux::OpaqueType) -> mlir::Type {
+    return llvmPointerType;
+  });
+  converter.addConversion([llvmPointerType](gpux::StreamType) -> mlir::Type {
+    return llvmPointerType;
+  });
+  converter.addConversion([llvmPointerType](gpux::DeviceType) -> mlir::Type {
+    return llvmPointerType;
+  });
+  converter.addConversion([llvmPointerType](gpux::ContextType) -> mlir::Type {
+    return llvmPointerType;
+  });
+
+  patterns.insert<
+      // clang-format off
+    ConvertGpuStreamCreatePattern,
+    ConvertGpuStreamDestroyPattern,
+    ConvertAllocOpToGpuRuntimeCallPattern,
+    ConvertDeallocOpToGpuRuntimeCallPattern
+  //   ConvertGpuModuleLoadPattern,
+  //   ConvertGpuModuleDestroyPattern,
+  //   ConvertGpuKernelGetPattern,
+  //   ConvertGpuKernelDestroyPattern,
+  //   ConvertGpuKernelLaunchPattern,
```

Comment:
I see these are place holders for now but any reason you left out patterns like ConvertMemcpyOpToGpuRuntimeCallPattern ?